src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -87,4 +97,34 @@ public int getTotalMessageCount() {
     PreparedQuery results = datastore.prepare(query);
     return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
+  
+  /** Returns the longest message length of all users. */
+  public int getLongestMessageCount() {
+    return longestMessage;
+  }
+  
+  /** Returns the total number of users that have posted. */
+  public int getTotalUserCount() {
+    return users.size();
+  }
+  
+  /** Returns the top three users that have posted on the website. */
+  public ArrayList<String> getTopUsers() {
+    ArrayList<String> topUsers = new ArrayList<String>(3);
+    int numTopUsers = 3;
+    String currTopUser = "";
+
+    while (numTopUsers >= 0) {
+      int maxPosts = 0;
+      for (String person : users.keySet()) {
+        if (users.get(person) > maxPosts && !topUsers.contains(person)) {
+          maxPosts = users.get(person);
+          currTopUser = person;
+        }
+      }
+      topUsers.add(currTopUser);
+      numTopUsers -= 1;
+    } 
+    return topUsers;
```

Comment:
The algorithm for this could be optimized a bit but it's fine for now.
However a comment about what you're trying to do ("Find the three users with the most posts") right before you start looping might make things a bit more obvious when reading through

Reply:
I figured my algorithm for this was bad...I wish to talk about this in our 1 on 1 because I found me coding this way to cause me issues in coding interviews and such in the past...either way, comment added!